backend/src/v1/services/report-search-service.spec.ts

```diff
@@ -0,0 +1,310 @@
+import createPrismaMock from 'prisma-mock';
+import { Prisma } from '@prisma/client';
+import { reportSearchService } from './report-search-service';
+
+const company1 = {
+  company_id: '4492feff-99d7-4b2b-8896-12a59a75d4e1',
+  company_name: 'Company Name 1',
+};
+const company2 = {
+  company_id: '4492feff-99d7-4b2b-8896-12a59a75d4e2',
+  company_name: 'Company Name 2',
+};
+
+const prismaClient: any = createPrismaMock({
+  pay_transparency_report: [
+    {
+      report_id: '4492feff-99d7-4b2b-8896-12a59a75d4e1',
+      create_date: '2024-04-19 21:46:53.876',
+      naics_code: '30',
+      reporting_year: 2024,
+      is_unlocked: true,
+      employee_count_range_id: '4492feff-99d7-4b2b-8896-12a59a75d4e1',
+      company_id: company1.company_id,
+    },
+    {
+      report_id: '4492feff-99d7-4b2b-8896-12a59a75d4e2',
+      create_date: '2023-04-19 21:46:53.876',
+      naics_code: '11',
+      reporting_year: 2023,
+      is_unlocked: false,
+      employee_count_range_id: '4492feff-99d7-4b2b-8896-12a59a75d4e9',
+      company_id: company2.company_id,
+    },
+    {
+      report_id: '4492feff-99d7-4b2b-8896-12a59a75d4e3',
+      create_date: '2022-04-19 21:46:53.876',
+      naics_code: '40',
+      reporting_year: 2021,
+      is_unlocked: false,
+      employee_count_range_id: '4492feff-99d7-4b2b-8896-12a59a75d4e5',
+      company_id: company2.company_id,
+    },
+  ],
+  pay_transparency_company: [company2, company1],
+});
+
+jest.mock('../prisma/prisma-client-readonly-replica', () => ({
+  __esModule: true,
+  ...jest.requireActual('../prisma/prisma-client-readonly-replica'),
+  default: {
+    pay_transparency_report: {
+      findMany: (args) => prismaClient.pay_transparency_report.findMany(args),
+      count: (args) => prismaClient.pay_transparency_report.count(args),
+    },
+  },
+}));
+
+describe('report-search-service', () => {
```

Comment:
Can there be some tests for invalid parameters like a key that we don't search on, an invalid object property name, an invalid operation, maybe a value that is expecting an int but instead got a string? Eg. [{key:"user_id", comparator:"bigger", value:"17"}]

Reply:
@goemen I just noticed this was done on line 64, sorry didn't notice that earlier.